Sources/Sentry/include/SentryTracerMiddleware.h

```diff
@@ -0,0 +1,41 @@
+#import "SentrySpan.h"
+#import "SentryTracer.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * Protocol used to intercept some of `SentryTracer` steps
+ * in order to modify its behavior and distribute concern to other classes.
+ */
+@protocol SentryTracerMiddleware <NSObject>
```

Comment:
`h`: I propose to rename this to `SentryTracerExtension`. It has three non optional methods `installForTracer`, `createAdditionalSpansForTrace` and `addMeasurements`.  This will allow us to also move app start measurements and the frame tracker measurements to extension classes.

As some tracer extensions could be interested in lifecycle events `SentryTracerExtension` should extend `TracerLifecycleEvents` which has only optional methods.

Reply:
I like the name very much.
I don't like the idea of dividing it, IMO it does not add value and make things more complicated.
Also, I don't like the assumption that every extension will need additional spans or measurements, INP for example will only require measurements.

Reply:
As discussed on a call, we will not create a `TracerLifecycleEvents` protocol. Instead, we do https://github.com/getsentry/sentry-cocoa/pull/2724/files#r1123098211.

We also remove the `@optional` so we don't have to check if a protocol implements a method in the tracer. We stick to the naming of `SentryTracerExtension`.